nodejs/toolchain.bzl

```diff
@@ -25,6 +25,31 @@ May be empty if the target_tool_path points to a locally installed node binary."
     },
 )
 
+def node_toolchain_providers(ctx, toolchain_info):
+    """Convert the toolchain_info struct to Providers returned by the toolchain.
+
+       This is exposed as part of the workaround for https://github.com/bazelbuild/bazel/issues/14009
+    """
+    return [
+        DefaultInfo(
+            files = depset(toolchain_info.tool_files),
+            runfiles = ctx.runfiles(files = toolchain_info.tool_files),
+        ),
+        platform_common.ToolchainInfo(
+            nodeinfo = NodeInfo(
```

Comment:
The types are unclear to me here: Are you converting from one provider to another, or just duplicating the same provider type?

I would assume that it's the same provider type, and so you don't actually need to create new instances of `NodeInfo` (or even `ToolchainInfo`).